File: BOJ/queue/boj11866.java
import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;

public class boj11866 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Queue<Integer> q = new LinkedList<>();

        int N = sc.nextInt();
        int K = sc.nextInt();

        for (int i = 1; i <= N; i++) {
            q.add(i);
        }
        StringBuilder sb = new StringBuilder();
        sb.append('<');

        while (q.size() > 1) {
            for (int i = 0; i < K - 1; i++) {
                int num = q.poll();
                q.offer(num);
            }
            sb.append(q.poll()).append(", ");
        }

        sb.append(q.poll()).append('>');

        System.out.println(sb);
    }
}

File: new-algorithm/src/kotlinalgo/queue/boj11866.kt
package kotlinalgo.queue

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.LinkedList
import java.util.Queue
import java.util.StringTokenizer

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val st = StringTokenizer(readLine())
    val sb = StringBuilder()

    val N = st.nextToken().toInt()
    val K = st.nextToken().toInt()

    val queue: Queue<Int> = LinkedList()

    for (i in 1..N) {
        queue.add(i)
    }

    sb.append('<')
    while (queue.size != 1) {
        for (i in 0 until K - 1) {
            queue.add(queue.poll())
        }
        sb.append(queue.poll()).append(", ")
    }
    sb.append(queue.poll()).append('>')

    println(sb)
}